cirq-core/cirq/ops/gate_features.py

```diff
@@ -36,20 +36,43 @@ class SupportsOnEachGate(raw_types.Gate, metaclass=abc.ABCMeta):
 
     def on_each(self, *targets: Union[raw_types.Qid, Iterable[Any]]) -> List[raw_types.Operation]:
         """Returns a list of operations applying the gate to all targets.
-
         Args:
-            *targets: The qubits to apply this gate to.
-
+            *targets: The qubits to apply this gate to. For single-qubit gates
+            this can be provided as varargs or a combination of nested
+            iterables. For multi-qubit gates this must be provided as an
+            `Iterable[Sequence[Qid]]`, where each sequence has `num_qubits`
+            qubits.
         Returns:
             Operations applying this gate to the target qubits.
-
         Raises:
-            ValueError if targets are not instances of Qid or List[Qid].
-            ValueError if the gate operates on two or more Qids.
+            ValueError if targets are not instances of Qid or Iterable[Qid].
+            ValueError if the gate qubit number is incompatible.
         """
+        operations: List[raw_types.Operation] = []
         if self._num_qubits_() > 1:
-            raise ValueError('This gate only supports on_each when it is a one qubit gate.')
-        operations = []  # type: List[raw_types.Operation]
+            if not targets:
```

Comment:
It might be more clear to handle the non-varargs case as the special case, and in that case reassign `targets`. Something like:
```python
if len(targets) == 1:
    if not isinstance(targets[0], Iterable):
        raise TypeError(...)
    t0 = list(targets[0])
    targets = [t0] if t0 and isinstance(t0[0], raw_types.Qid) else t0
for target in targets:
    if not isinstance(target, Sequence):
        ...
```